### ISSUE TITLE
[sheet] Show cupertino modal?

### ISSUE THREAD

**Issue by itsJoKr** [jamesblasco/modal_bottom_sheet#367]
I really like the direction _sheet_ is going and how it can work with Navigator2.

The only reason I use modal_bottom_sheet is for iOS-styled modals, which you can achieve with modal_bottom_sheet (but not in Navigator2-friendly way).  I didn't find any example of this with _sheet_. Is it possible to do it?

Example of what I'm talking about:
![image](https://github.com/jamesblasco/modal_bottom_sheet/assets/11093480/109a3916-a398-483c-92a8-698a07c311a4)

**Comment by cancng** [comment]
Same here, I'm trying to get this animation.

https://www.hackingwithswift.com/img/books/quick-start/swiftui/how-to-present-a-new-view-using-sheets-1.mp4

**Comment by bahadurh** [comment]
@cancng  were you able to create this animation? I m not able to create it :(

**Comment by cancng** [comment]
@bahadurh No, I couldn't. In React Native, this presentation type is pageSheet, but I couldn't find it in Flutter. 😦

**Comment by paldepind** [comment]
You can achieve this do this using `CupertinoSheetRoute` or `CupertinoSheetPage`. It's briefly alluded to [in the docs](https://github.com/jamesblasco/modal_bottom_sheet/tree/main/sheet#sheet-route).

I'm using go_router and was able to create an iOS-style modal by following [this example](https://github.com/jamesblasco/modal_bottom_sheet/blob/main/sheet/example/lib/examples/route/navigation/go_router.dart). The key thing is that you need to use `pageBuilder` in go_router to build your normal/non-modal pages with `CupertinoExtendedPage` (if you don't do this the underlying page won't animate into the background) and similarly you need to use `pageBuilder` to build you modal page with `CupertinoSheetPage`. Here's an example of what it looks like from my app: 

```dart
GoRoute(
  path: '/timer',
  pageBuilder: (BuildContext context, GoRouterState state) =>
      const CupertinoExtendedPage(child: StartTimerScreen()),
  routes: <RouteBase>[
    GoRoute(
      path: 'select-tags',
      pageBuilder: (BuildContext context, GoRouterState state) =>
          const CupertinoSheetPage(child: SelectTagsScreen()),
    ),
  ],
),
```

Additionally, if you build the example app and go to the "Route" tab you'll see a bunch of examples that showcase various ways to use an iOS-style modal. Many of these don't use go_router and thus show how to create the modal using the vanilla navigator API.

**Comment by hectorAguero** [comment]
as the 1.0 version, the package doesn't have a `fit` property exposed inside CupertinoSheetPage so the modal always use the full height of the screen.

I ended up making a copy of `CupertinoSheetPage`  and exposed the variable, not sure if there's a reason for that.

Also there is some issues with padding and Safearea

**Comment by lucasjinreal** [comment]
How to use previous screen push back without GoRouter? why have to stick with go route ...

**Comment by masal9pse** [comment]
Sample code is presented in [the official introduction of the sheet package](https://pub.dev/packages/sheet#sheet-route) and in [the above thread](https://github.com/jamesblasco/modal_bottom_sheet/issues/367#issuecomment-1806835188) with go_router.
Can anyone provide a sample code with [auto_route](https://pub.dev/packages/auto_route) that would result in similar behavior?
